lib/topic_view.rb

```diff
@@ -237,12 +237,7 @@ def published_time
 
   def image_url
     if @post_number > 1 && @desired_post.present?
-      if @desired_post.image_url.present?
-        @desired_post.image_url
-      elsif @desired_post.user
-        # show poster avatar
-        @desired_post.user.avatar_template_url.gsub("{size}", "200")
-      end
+      @desired_post.image_url
```

Comment:
I think we should be more clever here and try to always get an image.

```ruby
def image_url
  url = @desired_post&.image_url if @post_number > 1
  url || @topic.image_url
end
```

Reply:
Also, I think this is worth testing.

Reply:
:+1: agreed. Updated the logic and added some specs